js/client.js

```diff
@@ -368,6 +368,18 @@
 						}
 					}
 					this.send('/autojoin ' + autojoinIds.join(','));
+
+					var battleAutojoin = (Tools.prefs('battleautojoin') || '').split(',');
```

Comment:
This also might interact weirdly with opening multiple tabs; you probably don't want battles to get focused instantly.